be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala

```diff
@@ -470,4 +535,131 @@ sealed class ElectionValidator(dbActorRef: => AskableActorRef) extends MessageDa
+    *   the error to forward in case where the number of votes is incoherent.
+    * @return
+    *   GraphMessage: passes the rpcMessages to Right if successful else Left with pipeline error
+    */
+  private def checkNumberOfVotes(rpcMessage: JsonRpcRequest, channel: Channel, result: List[ElectionQuestionResult], dbActorRef: AskableActorRef, error: PipelineError): GraphMessage = {
+    val askLaoData = dbActorRef ? DbActor.ReadLaoData(channel)
+    Await.ready(askLaoData, duration).value match {
+      case Some(Success(DbActor.DbActorReadLaoDataAck(laoData: LaoData))) =>
+        val numberOfAttendees = laoData.attendees.size
+        val correct: Boolean =
+          result.forall(electionQuestionResult =>
+            (countNumberOfVotes(electionQuestionResult) <= numberOfAttendees) && allBallotCountsArePositive(electionQuestionResult)
+          )
+        if (correct) {
+          Right(rpcMessage)
+        } else {
+          Left(error)
+        }
+      case Some(Failure(ex: DbActorNAckException)) => Left(PipelineError(ex.code, s"validation of ResultElection failed, could not retrieve laoData : ${ex.message}", rpcMessage.getId))
+      case reply                                   => Left(PipelineError(ErrorCodes.SERVER_ERROR.id, s"ElectionValidator failed : unexpected DbActor reply '$reply'", rpcMessage.getId))
+    }
+  }
+
+  /** @param electionQuestionResult
+    *   The ElectionQuestionResult the ElectionValidator received.
+    * @return
+    *   The total number number of votes on that question.
+    */
+  private def countNumberOfVotes(electionQuestionResult: ElectionQuestionResult): Int = {
+    electionQuestionResult.result.map(_.count).sum
+  }
+
+  /** @param electionQuestionResult
+    *   The ElectionQuestionResult the ElectionValidator received.
+    * @return
+    *   True if all the ballot counts are greater that 0, false otherwise.
+    */
+  private def allBallotCountsArePositive(electionQuestionResult: ElectionQuestionResult): Boolean = {
+    electionQuestionResult.result.forall(electionBallotVotes => electionBallotVotes.count >= 0)
+  }
+
+  /** checks that the question ids received with the ResultElection are coherent with the ones received in the SetupElection
+    * @param rpcMessage
+    *   the rpc message to validate.
+    * @param questions
+    *   The list of ElectionQuestions that have been sent with the SetupElection.
+    * @param result
+    *   The list of ElectionQuestionResults the ElectionValidator received.
+    * @param electionId
+    *   The electionId to compute the election question ids.
+    * @param error
+    *   The error to forward in case where ballot options are incoherent.
+    * @return
+    *   GraphMessage: passes the rpcMessages to Right if successful else Left with pipeline error.
+    */
+  private def checkResultQuestionIds(
+      rpcMessage: JsonRpcRequest,
+      questions: List[ElectionQuestion],
+      result: List[ElectionQuestionResult],
+      electionId: Hash,
```

Comment:
You do not use this parameter inside the function. Thus, you should remove it from the function definition